vscode debug config for unit tests

ostreifel opened the pull request.

ostreifel committed:
vscode debug config for unit tests

cschleiden commented on tsconfig.json:
```diff
@@ -1,12 +1,17 @@
 {
     "compilerOptions": {
-        "module": "amd",
+        "module": "umd",
```
why this change? Since VSTS uses AMD I'd like to keep the module format consistent

ostreifel replied:
Node requires special configuration to use amd. So I went with the compromise module configuration. Alternatively we could have a separate tsconfig file for tests to use commonjs module configuration

cschleiden commented on tsconfig.json:
```diff
@@ -1,12 +1,17 @@
         "!**/*.tests.ts",
         "./typings/index.d.ts"
+    ],
+    "files": [
```
I think these should not be here, any tool that supports `filesGlob` should be able to expand them

cschleiden commented on .gitignore:
```diff
@@ -2,4 +2,6 @@ node_modules
\ No newline at end of file
+build
+test
+.baseDir.ts
```
when is this added?

ostreifel committed:
Separate tsconfig file for tests

ostreifel commented:
Updated to have tests use a separate tsconfig file so that ./build uses amd and ./test uses umd (Needs to be both amd compatible for grunt test and commonjs compatible when debugging with node). Also removed duplicate type definitions that break the build when the '@types' directory is added in node_modules
